GVRf/Framework/jni/objects/components/camera_rig_jni.cpp

```diff
@@ -197,16 +195,22 @@ Java_org_gearvrf_NativeCameraRig_setCameraSeparationDistance(
     camera_rig->set_camera_separation_distance(distance);
 }
 
-JNIEXPORT jfloat JNICALL
+JNIEXPORT jfloatArray JNICALL
 Java_org_gearvrf_NativeCameraRig_getFloat(JNIEnv * env,
         jobject obj, jlong jcamera_rig, jstring key) {
     std::shared_ptr<CameraRig> camera_rig = *reinterpret_cast<std::shared_ptr<
             CameraRig>*>(jcamera_rig);
     const char* char_key = env->GetStringUTFChars(key, 0);
     std::string native_key = std::string(char_key);
-    jfloat camera_rig_float = camera_rig->getFloat(native_key);
     env->ReleaseStringUTFChars(key, char_key);
-    return camera_rig_float;
+    try {
+        jfloat value = camera_rig->getFloat(native_key);
+        jfloatArray result_float = env->NewFloatArray(1);
+        env->SetFloatArrayRegion(result_float, 0, 1, &value);
+        return result_float;
+    } catch (const std::string & err) {
+        return NULL;
```

Comment:
I think, overall is fine. One small concern is about performance. I know that you may not want to use NaN or Infinity as illegal state flag. But I think this way may be a little bit worse in terms of performance since you are going to create an array each time.

I mean, once if this one or similar methods get called so many times, it may cause a little performance degradation. Of course this is not the case in our sample apps right now and I don't see if there is any such case in the future. However compared to using NaN, this is not necessarily safer but still a little bit worse in performance.